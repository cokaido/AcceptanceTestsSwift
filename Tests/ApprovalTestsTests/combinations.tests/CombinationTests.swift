import XCTest
#if os(OSX)
@testable import ApprovalTests_Swift
#elseif os(iOS)
@testable import ApprovalTests_iOS
#endif

class CombinationTests: XCTestCase {

    func threeParams(_ in1: Int, _ in2: Int, _ in3: Int ) -> Int {
        return in1 + in2 + in3
    }

    func twoIntParams(_ in1: Int, _ in2: Int ) -> Int {
        return in1 * in2
    }

    func twoFloatParams( _ in1: Float, _ in2: Float ) -> Float {
        return in1 * in2
    }

    func doubleInt(_ x: Int) -> Int {
        return 2 * x
    }

    func doubleFloat(_ x: Float) -> Float {
        return 2 * x
    }

    func threeMixedParams( _ anInt: Int, _ aFloat: Float, _ aString: String ) -> String {
        return String("\(anInt) \(aFloat) \(aString)")
    }

    func testCombinationsSingleParameterInt() {
        try! CombinationApprovals.verifyAllCombinations(doubleInt, [1,2,3,4,5])
    }

    func testCombinationsSingleParameterFloat() {
        try! CombinationApprovals.verifyAllCombinations(doubleFloat, [1.1, 2.2, 3.3, 4.4, 5.5])
    }

    func testCombinationsTwoParametersInt() {
        try! CombinationApprovals.verifyAllCombinations(twoIntParams, [1, 2], [3, 4])
    }

    func testCombinationsMixedParameters() {
        try! CombinationApprovals.verifyAllCombinations(threeMixedParams, [1,2,3], [4.4, 5.5], ["String1", "String2", "String3", "String4"])
    }
    
    func testCombinationsFourParameters() {
        let inCall: (Int, Int, Int, Int) -> String = {
            let sum = $0 + $1 + $2 + $3
            return String(sum)
        }

        try! CombinationApprovals.verifyAllCombinations(inCall, [1,2,3], [4,5], [6,7], [8,9])
    }

    func testCombinationsFiveParameters() {
        let inCall: (Int, Int, Int, Int, Int) -> String = {
            let sum = $0 + $1 + $2 + $3 + $4
            return String(sum)
        }
        
        try! CombinationApprovals.verifyAllCombinations(inCall, [1,2,3], [4,5], [6,7], [8,9], [10,11])
    }

    func testCombinationsSixParameters() {
        let inCall: (Int, Int, Int, Int, Int, Int) -> String = {
            let sum = $0 + $1 + $2 + $3 + $4 + $5
            return String(sum)
        }
        
        try! CombinationApprovals.verifyAllCombinations(inCall, [1,2,3], [4,5], [6,7], [8,9], [10,11], [12,13])
    }

    func testCombinationsSevenParameters() {
        let inCall: (Int, Int, Int, Int, Int, Int, Int) -> String = {
            let sum = $0 + $1 + $2 + $3 + $4 + $5 + $6
            return String(sum)
        }
        
        try! CombinationApprovals.verifyAllCombinations(inCall, [1,2,3], [4,5], [6,7], [8,9], [10,11], [12,13], [14,15])
    }

    func testCombinationsEightParameters() {
        let inCall: (Int, Int, Int, Int, Int, Int, Int, Int) -> String = {
            let sum = $0 + $1 + $2 + $3 + $4 + $5 + $6 + $7
            return String(sum)
        }
        
        try! CombinationApprovals.verifyAllCombinations(inCall, [1,2,3], [4,5], [6,7], [8,9], [10,11], [12,13], [14,15], [16,17])
    }

    func testCombinationsNineParameters() {
        let inCall: (Int, Int, Int, Int, Int, Int, Int, Int, Int) -> String = {
            let sum = $0 + $1 + $2 + $3 + $4 + $5 + $6 + $7 + $8
            return String(sum)
        }
        
        try! CombinationApprovals.verifyAllCombinations(inCall, [1,2,3], [4,5], [6,7], [8,9], [10,11], [12,13], [14,15], [16,17], [18,19])
    }

}
