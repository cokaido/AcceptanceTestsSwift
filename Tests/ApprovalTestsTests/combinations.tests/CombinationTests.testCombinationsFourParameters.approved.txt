[1, 4, 6, 8] -> 19
[1, 4, 6, 9] -> 20
[1, 4, 7, 8] -> 20
[1, 4, 7, 9] -> 21
[1, 5, 6, 8] -> 20
[1, 5, 6, 9] -> 21
[1, 5, 7, 8] -> 21
[1, 5, 7, 9] -> 22
[2, 4, 6, 8] -> 20
[2, 4, 6, 9] -> 21
[2, 4, 7, 8] -> 21
[2, 4, 7, 9] -> 22
[2, 5, 6, 8] -> 21
[2, 5, 6, 9] -> 22
[2, 5, 7, 8] -> 22
[2, 5, 7, 9] -> 23
[3, 4, 6, 8] -> 21
[3, 4, 6, 9] -> 22
[3, 4, 7, 8] -> 22
[3, 4, 7, 9] -> 23
[3, 5, 6, 8] -> 22
[3, 5, 6, 9] -> 23
[3, 5, 7, 8] -> 23
[3, 5, 7, 9] -> 24
